ajax/getAbstractRightDropdownValue.php

```diff
@@ -31,12 +31,14 @@
  * ---------------------------------------------------------------------
  */
 
-include('../inc/includes.php');
+include(__DIR__ . '/../inc/includes.php');
 
 header("Content-Type: application/json; charset=UTF-8");
 Html::header_nocache();
 Session::checkLoginUser();
 
-$search = $_POST['searchText'] ?? "";
-
-echo json_encode(ShareDashboardDropdown::fetchValues($search));
+function show_rights_dropdown(string $class)
+{
+    $search = $_POST['searchText'] ?? "";
+    echo json_encode($class::fetchValues($search));
+}
```

Comment:
Logic used in other parts of GLPI is reversed, see `dropdown.common.php`.

To use same kind of logic, you should `include(__DIR__ . '/../inc/includes.php');` in the caller file, then define `$class = ShareDashboardDropdown::class;`, then include current file that execute its code.

Reply:
Wouldn't this be less convenient as it makes the child code run before the parent ?
Even thought it isn't really a parent/child inheritance situation as it isn't OOP code, it seems more intuitive to run the "parent" code first.

If you consider the rights checks for example, shouldn't you do the common check (is the user logged in ?) before the specific ones ?